### PR TITLE
Add expression assembler

### PR DESCRIPTION
**Pull request by filiatra**
First addition of expression assembler

### CODE REVIEW

File: examples/poisson2_example.cpp [hunk 1]
```diff
@@ -1,238 +1,185 @@
...
-
-    gsCmdLine cmd( "Solves Poisson's equation with an isogeometric discretization." );
-    cmd.addString( "p", "pde", "File containing a poisson PDE (.xml)", fn_pde );
-    cmd.addSwitch( "nitsche", "Use the Nitsche's method for Dirichlet sides", Nitsche );
-    cmd.addSwitch( "discGalerkin", "Use Discontiouous Galerkin method for patch interfaces", dG );
-    cmd.addSwitch( "plot", "Plot result in ParaView format", plot );
-    cmd.addInt( "s", "plotSamples", "Number of sample points to use for plotting", plot_pts );
+    int numRefine  = 5;
+    int numElevate = 0;
+    bool last = false;
```
[the remark this answers is not in the document]
Comment (filiatra) [reply examples/poisson2_example.cpp]: With the current settings it runs fast enough as it is, we can look again in CDash later and change this if needed..

File: examples/poisson2_example.cpp [hunk 2]
```diff
@@ -1,238 +1,185 @@
...
-            bases[j].uniformRefine();
+        solver.compute( A.matrix() );
+        solVector = solver.solve(A.rhs());
 
-    gsInfo << "Discrete space 0: "<< bases[0] << "\n";
+        gsInfo<< "." <<std::flush; // Linear solving done
 
+        l2err[r]= math::sqrt( ev.integral( (u_ex - u_sol).sqNorm() * meas(G) ) );
+        h1err[r]= l2err[r] +
+        math::sqrt(ev.integral( ( grad(u_ex) - grad(u_sol)*jac(G).inv() ).sqNorm() * meas(G) ));
```
[the remark this answers is not in the document]
Comment (filiatra) [reply examples/poisson2_example.cpp]: u_ex is a given known solution defined on physical space, so no transformation via G is needed.

File: src/gsAssembler/gsExpressions.h [hunk 3]
```diff
@@ -0,0 +1,2749 @@
...
+                     Eigen::Matrix<T,-1,-1> & hessian)
+{
+    const index_t sz = dim*(dim+1)/2;
+    const gsAsConstMatrix<T> ders(secDers.data(), sz, secDers.size() / sz );
+    hessian.resize(dim*dim, ders.cols() );
+
+    switch ( dim )
+    {
+    case 1:
+        hessian = secDers;
```
[the remark this answers is not in the document]
Comment (filiatra) [reply src/gsAssembler/gsExpressions.h]: there is no warning, it is used already outside the switch

File: src/gsAssembler/gsExpressions.h [hunk 4]
```diff
@@ -0,0 +1,2749 @@
...
+        break;
+    default:
+        break;
+    }
+}
+
+namespace expr
+{
+
+#if(__cplusplus >= 201402L) // c++14
```
[the remark this answers is not in the document]
Comment (filiatra) [reply src/gsAssembler/gsExpressions.h]: Maybe it is not needed, but I just hope that optimizations might pop in in everything is auto..

File: src/gsAssembler/gsExpressions.h [hunk 5]
```diff
@@ -0,0 +1,2749 @@
...
+template<class T> class meas_expr;
+template<class E> class inv_expr;
+template<class E> class tr_expr;
+template<class E> class temp_expr;
+//bool = E1::ScalarValued && E2::ScalarValued
+template<class E1, class E2, bool = E1::ColBlocks> class mult_expr
+{using E1::GISMO_ERROR_mult_expr_has_invalid_template_arguments;};
+
+template<class T> class gsFeVariable;
+template<class T> class gsFeSolution;
```
[the remark this answers is not in the document]
Comment (filiatra) [reply src/gsAssembler/gsExpressions.h]: lol...

File: src/gsAssembler/gsExpressions.h [hunk 6]
```diff
@@ -0,0 +1,2858 @@
...
+    { }
+};
+
+
+/*
+   Expression representing a function given by a vector of
+   coefficients in a gsFeSpace.
+
+   Typically it used for accessing the solution of a boundary-value
+   problem.  space
```
Comment (stefantakacs) [inline src/gsAssembler/gsExpressions.h]: rm "space"

File: src/gsAssembler/gsExpressions.h [hunk 7]
```diff
@@ -0,0 +1,2858 @@
...
+        else
+            return _v.colVar();
+        */
+    }
+
+    void print(std::ostream &os) const { _u.print(os); os<<"*"; _v.print(os); }
+};
+
+/*
+  Expression for multip
```
Comment (stefantakacs) [inline src/gsAssembler/gsExpressions.h]: Do these things need DOXYGEN comments?

Comment (stefantakacs) [reply src/gsAssembler/gsExpressions.h]: no

**Pull request merged**